rubi/rubi/data/data.py

```diff
@@ -1,72 +1,46 @@
+import logging as log
 from subgrounds import Subgrounds
-from rubi.data.sources.aid import AidData, SuperAidData
-from rubi.data.sources.market import MarketData
-from rubi.data.sources.helper import Gas, Price, networks 
-from rubi.data.processing.user import User, SuperUser
-from rubi.data.processing.aid import AidProcessing
+from typing import Union, List, Optional, Dict, Type, Any, Callable
 
-# TODO this needs significant love.
-class Data: 
-    """this class acts as the main access point to a variety of data and data tooling for the Rubicon protocol. it acts as a data processing layer built using the subgrounds library and the subgraphs maintained at the follwing repo: https://github.com/RubiconDeFi/rubi-subgraphs
-    """
-
-    def __init__(self, super=False): 
-        """constructor method. creates a subgrounds object that is then used to initialize a variety of data sources and tooling
+from rubi.data.sources import (
+    MarketData,
+)
 
-        :param super: whether the data object is a super data object, defaults to False
-        :type super: bool, optional
-        """
-        self.subgrounds = Subgrounds()
-        self.price = Price()
-        self.networks = networks
+from rubi.network import (
+    Network,
+)
 
-        # initialize the data sources
-        self.market_optimism = MarketData(self.subgrounds, 10)
-        self.market_optimism_goerli = MarketData(self.subgrounds, 420)
-        
-        # initialize the data processing if the data object is not a super data object
-        if not super:
-            self.user = User(self.subgrounds, self.market_optimism)
-            self.market_aid_optimism = AidData(self.subgrounds, 10) 
-            self.market_aid_optimism_goerli = AidData(self.subgrounds, 420)
-
-            # TODO: we most likely want to rearrange this so that the processing class uses the AidData class as a parent class
-            # this issue here is that we are using a parent/child relationship already to distinguish between a web3 enabled data object and a non web3 enabled data object
-            # we should probably simplify this to just use a conditional statement to determine whether or not we are using a web3 enabled data object
-            # if anyone has a lot of thoughts here please feel free to open an issue on the repo :)
-            self.market_aid_optimism_processing = AidProcessing(self.subgrounds, 10, self.market_aid_optimism)
-            self.market_aid_optimism_goerli_processing = AidProcessing(self.subgrounds, 420, self.market_aid_optimism_goerli)
+# TODO: in the future, we will need to handle the need to query the decentralized network, requiring a wallet + GRT
 
-class SuperData(Data):
-    """this class acts as an extension of the data class with additional functionality that is enabled by being connected to a node. 
+class Data: 
```

Comment:
I question if this class is necessary.

The only thing it seems to do is hold an instance of `Subgrounds`.

We could have this instance on the `Network` and then the client will also just have a market data instance.

For me the analogy is:

Subgrounds = w3  

market contract = market data

(let me know if this makes sense :joy:)

Reply:
i agree with you here. in my head this class was really just here in case we decided to handle multiple networks through a single client, then the data class would manage all of those networks.

i think there may be a point in the future where we bring this back to help facilitate some more complex data processes, but for now there is no need for it.